doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -510,6 +510,19 @@ Requests to the environment
 
 .. seealso:: Section :ref:`locating-notations`
 
+.. _printing-flags:
+
+Printing flags
+-------------------------------
+
+.. flag:: Fast Name Printing
+
+   When turned on, |Coq| uses an asymptotically faster algorithm for the
+   attribution of non-ambiguous names of bound variables while printing terms.
+   While faster, it is also less clever and results in a typically less elegant
+   display. This flag is not enabled by default, because as Ltac observes bound
+   names, turning it on can break existing proof scripts.
+
```

Comment:
Perhaps you could (briefly) be more explicit about the benefit of and when to use this feature.  It's unexpected to me that there is a printing operation where there is a performance issue.  Maybe you could include/adapt more of the info from the PR description?

> "because as Ltac observes bound names, turning it on can break existing proof scripts."

I don't quite get this.  Do you mean that the bound names appear in the proof scripts?  How the names "observed"?  Is there a simple example of the problem? (BTW, I've not used Ltac.)

Reply:
> It's unexpected to me that there is a printing operation where there is a performance issue. 

It's a pervasive issue. I keep seeing this hotspot in virtually any non-trivial printing. The algorithm is at worst cubic and quadractic on average, you don't need a lot of bound variables to see the effects.

> Is there a simple example of the problem?

Typical tactic observing printing is `destruct n` where `n` is a bound name, e.g.
```
Goal forall n : nat, n = n.
Proof.
destruct n.
```


Change the bound name and this will break the script.

Reply:
I didn't try your example with the new code, but how about adding "You can see the difference by running the following script with and without the option: ..." if that's accurate.

Reply:
"because as Ltac observes bound names, turning it on can break existing proof scripts"

It can break any script that applies an Ltac tactic, such as "easy"?  Is that the only way it breaks?  Seems like this isn't enough info for the user to know whether they can safely use the flag.

Reply:
Unluckily, I believe that there is no way to know whether you can safely use the flag apart from calling your tactic and seeing whether it broke. Most tactics rely ultimately on some mechanism that might observe bound names, e.g. even the lower-level variants of intro used virtually everywhere.